### ISSUE TITLE
FGVC caltech101 results

### ISSUE THREAD

**Issue by khawar-islam** [3neutronstar/GuidedMixup#6]
Hello @3neutronstar 

Thanks for the FGVC code, I am currently training resnet18 on caltech101 via guided-ap and it is very hard to reproduce same results mentioned in the paper. do you have any idea? What should i do?

Training Command:

`python main.py train --dataset caltech101 "$@" --wandb False --train_mode guided-ap --condition greedy --mix_prob 0.8  --model resnet18`

Training Log
==============================
```
{'mode': 'train', 'seed': 1, 'deterministic': True, 'model': 'resnet18', 'device': 'cuda', 'batch_size': 16, 'num_workers': 4, 'weight_decay': 0.0005, 'lr': 0.01, 'epochs': 300, 'dataset': 'caltech101', 'no_verbose': False, 'training_verbose': False, 'warmup': 0, 'cooldown': 0, 'detect_anomaly': False, 'eval_loc': False, 'compile': False, 'apex': False, 'use_fp16': False, 'ddp': False, 'local_rank': 0, 'labels_per_class': None, 'wandb': False, 'criterion': 'crossentropy', 'optimizer': 'sgd', 'nesterov': False, 'momentum': 0.9, 'lr_scheduler': 'step', 'gamma': 0.1, 'lr_steps': [150, 225], 'mixup_type': 'plain', 'cutout': False, 'augmix': False, 'randaug': False, 'train_mode': 'guided-ap', 'mix_prob': 0.8, 'mix_prob_scheduling': False, 'condition': 'greedy', 'm_part': 16, 'distance_metric': 'l2', 'blur': 7, 'blur_sigma': 3, 'saliency_normalization': 'sumto1', 'improve': None, 'blur_random': False, 'blur_sigma_random': False, 'm_block_size': [], 'interpolate_mode': 'nearest', 'label_mixing': 'saliency_label', 'clean_lam': 0, 'mixup_epochs': 0, 'fgvc_resize': False, 'memo': '', 'dataset_path': None, 'corruption_dataset_path': None, 'gpu_ids': '0'}
==============================
Number of classes : 101
Resize shape: 256 crop shape: 224
```

**Last Epochs**

```
[07-29 04:08:27] [eval] [299 epoch] Loss: 13.6523 | top1: 15.8986 | top5: 32.7765 [Time Left]  0h  0m 23s
Epoch: [300][  0/434]   Time  0.033 ( 0.033)    Loss 1.4880e-02 (1.4880e-02)
Epoch: [300][217/434]   Time  0.049 ( 0.047)    Loss 1.4066e+00 (1.3730e+00)
[07-29 04:08:47] [train] Lr: 1.0000e-04 | Loss: 1.3763 | time: 20.199
[07-29 04:08:50] [eval] [300 epoch] Loss: 13.9705 | top1: 16.8779 | top5: 32.6613 [Time Left]  0h  0m  0s
Learning Time: 7065.9705s
[07-29 04:08:50] [Best Valid Accuracy] 41.36 [Saved Time] caltech101_resnet18_guided-ap_pgreedy_07-29_14-11-03
```

**Comment by 3neutronstar** [comment]
Following results is from mine... (Guided-SR).
Your validation loss is quite different compared with my results. I hope you should review with your validation data or results.
```
1236 Epoch: [299][  0/434] Time  0.140 ( 0.140) Loss 1.1079e+00 (1.1079e+00)
1237 Epoch: [299][217/434] Time  0.122 ( 0.126) Loss 1.4439e+00 (8.0990e-01)
1238 [12-18 07:09:34] [train] Lr: 1.0000e-04 | Loss: 0.7452 | time: 54.582
1239 [12-18 07:09:48] [eval] [299 epoch] Loss: 0.4128 | top1: 89.9770 | top5: 97.0046 [Time Left]  0h  1m 27s
1240 Epoch: [300][  0/434] Time  0.155 ( 0.155) Loss 8.0243e-03 (8.0243e-03)
1241 Epoch: [300][217/434] Time  0.100 ( 0.101) Loss 4.0939e-03 (7.5829e-01)
1242 [12-18 07:10:31] [train] Lr: 1.0000e-04 | Loss: 0.7552 | time: 43.053
1243 [12-18 07:10:37] [eval] [300 epoch] Loss: 0.4206 | top1: 89.6889 | top5: 97.1198 [Time Left]  0h  0m  0s
1244 Learning Time: 26087.6024s
1245 [12-18 07:10:37] [Best Valid Accuracy] 90.67 [Saved Time] caltech101_resnet18_guided-sr_12-18_11-55-38
```

Sorry for editing, if you undergo the same circumstances, I hope you would add `clean_lam` option as `1`.

I would check the argument that you gave me again. For me, trainings are going well.

**Comment by khawar-islam** [comment]
The dataset and everything is same as you mentioned in your code.

`python main.py train --dataset caltech101 "$@" --wandb False --train_mode guided-ap --condition greedy --mix_prob 0.8  --model resnet18 --clean_lam 1`


`{'mode': 'train', 'seed': 1, 'deterministic': True, 'model': 'resnet18', 'device': 'cuda', 'batch_size': 16, 'num_workers': 4, 'weight_decay': 0.0005, 'lr': 0.01, 'epochs': 300, 'dataset': 'caltech101', 'no_verbose': False, 'training_verbose': False, 'warmup': 0, 'cooldown': 0, 'detect_anomaly': False, 'eval_loc': False, 'compile': False, 'apex': False, 'use_fp16': False, 'ddp': False, 'local_rank': 0, 'labels_per_class': None, 'wandb': False, 'criterion': 'crossentropy', 'optimizer': 'sgd', 'nesterov': False, 'momentum': 0.9, 'lr_scheduler': 'step', 'gamma': 0.1, 'lr_steps': [150, 225], 'mixup_type': 'plain', 'cutout': False, 'augmix': False, 'randaug': False, 'train_mode': 'guided-ap', 'mix_prob': 0.8, 'mix_prob_scheduling': False, 'condition': 'greedy', 'm_part': 16, 'distance_metric': 'l2', 'blur': 7, 'blur_sigma': 3, 'saliency_normalization': 'sumto1', 'improve': None, 'blur_random': False, 'blur_sigma_random': False, 'm_block_size': [], 'interpolate_mode': 'nearest', 'label_mixing': 'saliency_label', 'clean_lam': 1.0, 'mixup_epochs': 0, 'fgvc_resize': False, 'memo': '', 'dataset_path': None, 'corruption_dataset_path': None, 'gpu_ids': '0'}`

Still getting same results, do you have any hint? I didn't change anything, same dataset and same configuration

[07-30 10:58:44] [train] Lr: 1.0000e-04 | Loss: 1.4191 | time: 20.564
[07-30 10:58:47] [eval] [299 epoch] Loss: 14.9570 | top1: 9.6774 | top5: 23.0991 [Time Left]  0h  0m 23s
Epoch: [300][  0/434]   Time  0.033 ( 0.033)    Loss 1.6744e-03 (1.6744e-03)
Epoch: [300][217/434]   Time  0.049 ( 0.047)    Loss 1.4653e+00 (1.4409e+00)
[07-30 10:59:08] [train] Lr: 1.0000e-04 | Loss: 1.4222 | time: 20.613
[07-30 10:59:11] [eval] [300 epoch] Loss: 18.1265 | top1: 5.5876 | top5: 18.4908 [Time Left]  0h  0m  0s
Learning Time: 7113.2537s
[07-30 10:59:11] [Best Valid Accuracy] 38.31 [Saved Time] caltech101_resnet18_guided-ap_pgreedy_07-30_09-00-37

**Comment by khawar-islam** [comment]
Dear @3neutronstar if you have some hint, please let me know i am waiting for this: id you change something like learning rate? scheduler? image size?

**Comment by 3neutronstar** [comment]
```
Epoch: [299][  0/434]   Time  0.219 ( 0.219)    Loss 1.2960e+00 (1.2960e+00)
Epoch: [299][217/434]   Time  0.216 ( 0.202)    Loss 1.5690e+00 (1.2383e+00)
[07-30 09:02:33] [train] Lr: 1.0000e-04 | Loss: 1.2417 | time: 87.581
[07-30 09:02:40] [eval] [299 epoch] Loss: 0.3894 | top1: 90.4378 | top5: 97.2926 [Time Left]  0h  1m 35s
Epoch: [300][  0/434]   Time  0.131 ( 0.131)    Loss 3.3578e-03 (3.3578e-03)
Epoch: [300][217/434]   Time  0.217 ( 0.202)    Loss 1.5089e+00 (1.2663e+00)
[07-30 09:04:08] [train] Lr: 1.0000e-04 | Loss: 1.2450 | time: 87.797
[07-30 09:04:15] [eval] [300 epoch] Loss: 0.4234 | top1: 89.8041 | top5: 96.7742 [Time Left]  0h  0m  0s
Learning Time: 28717.1590s
[07-30 09:04:18] [Best Valid Accuracy] 91.07 [Saved Time] caltech101_resnet18_guided-ap_pgreedy_07-30_13-05-22
```

After you asked me to recheck the implementation, I reproduced my result yesterday and found out that there was no problem.

I've never change any lr, scheduler or image size.
Here is the configuration that I used.
```
{'mode': 'train', 'seed': 1, 'deterministic': True, 'model': 'resnet18', 'device': 'cuda', 'batch_size': 16, 'num_workers': 4, 'weight_decay': 0.0005, 'lr': 0.01, 'epochs': 300, 'dataset': 'caltech101', 'no_verbose': False, 'training_verbose': False, 'warmup': 0, 'cooldown': 0, 'detect_anomaly': False, 'eval_loc': False, 'compile': False, 'apex': False, 'use_fp16': False, 'ddp': False, 'local_rank': 0, 'labels_per_class': None, 'wandb': False, 'criterion': 'crossentropy', 'optimizer': 'sgd', 'nesterov': True, 'momentum': 0.9, 'lr_scheduler': 'step', 'gamma': 0.1, 'lr_steps': [150, 225], 'mixup_type': 'plain', 'cutout': False, 'augmix': False, 'randaug': False, 'train_mode': 'guided-ap', 'mix_prob': 0.8, 'mix_prob_scheduling': False, 'condition': 'greedy', 'm_part': 16, 'distance_metric': 'l2', 'blur': 7, 'blur_sigma': 3, 'saliency_normalization': 'sumto1', 'improve': None, 'blur_random': False, 'blur_sigma_random': False, 'm_block_size': [], 'interpolate_mode': 'nearest', 'label_mixing': 'saliency_label', 'clean_lam': 1.0, 'mixup_epochs': 0, 'fgvc_resize': False, 'memo': '', 'dataset_path': None, 'corruption_dataset_path': None, 'gpu_ids': '0'}
``

**Comment by khawar-islam** [comment]
Hello @3neutronstar In your above answer you have reported  [07-30 09:04:18] [Best Valid Accuracy] **91.07** [Saved Time] caltech101_resnet18_guided-ap_pgreedy_07-30_13-05-22 and in your paper the number is **91.32**. there is a little difference and I can understand due its pytorch version or maybe some settings. I have proposed a new data augmentation strategy whose result is 91.30. Do you have any idea what should i do because your results is 91.32?

**Comment by 3neutronstar** [comment]
Your message was as follows:
Hello [@3neutronstar](https://github.com/3neutronstar) In your above answer you have reported [07-30 09:04:18] [Best Valid Accuracy] 91.07 [Saved Time] caltech101_resnet18_guided-ap_pgreedy_07-30_13-05-22 and in your paper the number is 91.32. there is a little difference and I can understand due its pytorch version or maybe some settings. I have proposed a new data augmentation strategy whose result is 91.30. Do you have any idea what should i do because your results is 91.32?

My comment:
Now I could not access my server anymore (I've already graduated.), therefore I could not run or reproduce the code actively anymore. The result that I commented above is one of my effort for you. However, to convince you, please refer the following result, which is the better result compared to the official result (91.32) conducted about 2 years ago. The hyper-parameter setting is the same.
```
1261 Epoch: [300][  0/434] Time  4.767 ( 4.767) Loss 1.5613e+00 (1.5613e+00)
1262 Epoch: [300][217/434] Time  0.376 ( 0.425) Loss 1.3752e+00 (1.2569e+00)
1263 [12-14 01:08:53] [train] Lr: 1.0000e-04 | Loss: 1.2619 | time: 172.915
1264 [12-14 01:09:20] [eval] [300 epoch] Loss: 0.3860 | top1: 90.8410 | top5: 97.6959 [Time Left]  0h  0m  0s
1265 Learning Time: 45833.7809s
1266 Model Performance:  {'loss': 0.3767184060633457, 'accuracy': 91.3594470046083, 'top5': 97.6958525345622}
1267 Load Best Model Complete
1268 [12-14 01:09:50] [Expected Calibration Error] 4.97 [NLL Loss] 3.7672e-01
1269 [12-14 01:09:52] [Best Valid Accuracy] 91.36 [Saved Time] caltech101_resnet18_guided-ap_pgreedy_c_12-13_12-25-11
```

**Comment by diffuseMix** [comment]
@3neutronstar thank you very much for your effort for me and i respect that, i can totally understand your point.
